### PR TITLE
Add missing information for iOS

### LINKED ISSUE #149: Documentation for gelocation plugin should be updated to meet new apple rules about location access
# Feature Request

## Motivation Behind Feature

Apple identified some issues without specified NSLocationAlwaysUsageDescription in info,plist, so the documentation for geolocation plugin is missed some informations, please update the documentation and add these rules to it:

edit-config target="NSLocationAlwaysAndWhenInUseUsageDescription" file="*-Info.plist" mode="merge">
          string>Need location access to find things nearby. /string>
      /edit-config>
       edit-config target="NSLocationWhenInUseUsageDescription" file="*-Info.plist" mode="merge">
       string>Need location access to find things nearby. /string>
      /edit-config>
      edit-config target="NSLocationAlwaysUsageDescription" file="*-Info.plist" mode="merge">
       string>Need location access to find things nearby./string>
     /edit-config>

(*** please note that i removed the start of codes because github not allowed these, so please add < in the first of each code)

Apple will send an email about issues found if you don't add the previous rules to info.plist.

Message from apple: 
Missing Purpose String in Info.plist File - Your app's code references one or more APIs that access sensitive user data. The app's Info.plist file should contain a NSLocationAlwaysUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. Starting Spring 2019, all apps submitted to the App Store that access user data will be required to include a purpose string. If you're using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. You can contact the developer of the library or SDK and request they release a version of their code that doesn't contain the APIs. Learn more (https://developer.apple.com/documentation/uikit/core_app/protecting_the_user_s_privacy).

## Feature Description

Please add these rules to the documentation:
/edit-config target="NSLocationAlwaysAndWhenInUseUsageDescription" file="*-Info.plist" mode="merge">
          string>Need location access to find things nearby. /string>
      /edit-config>
       edit-config target="NSLocationWhenInUseUsageDescription" file="*-Info.plist" mode="merge">
       string>Need location access to find things nearby. /string>
      /edit-config>
      edit-config target="NSLocationAlwaysUsageDescription" file="*-Info.plist" mode="merge">
       string>Need location access to find things nearby./string>
     /edit-config>

(*** please note that i removed the start of codes because github not allowed these, so please add < in the first of each code)

### PR DESCRIPTION
**Pull request by mosabab**
closes #149

### Platforms affected
IOS

### Motivation and Context

Add missing information regarding this plugin for ios geolocation requirements. 

### Description

Add missing information, by describes the reason that the app accesses the user's location in different situations.

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change
- [x] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [x] I've updated the documentation if necessary

### CODE REVIEW

**Comment by mosababubakr** [comment]
Hello @breautek , I also update this documents with missing information, please review it. Thanks

File: README.md [hunk 1]
```diff
@@ -160,15 +160,27 @@ error, the `geolocationError` callback is passed a
 
  This plugins requires the following usage description:
 
- * `NSLocationWhenInUseUsageDescription` describes the reason that the app accesses the user's location. 
+ * `NSLocationWhenInUseUsageDescription` describes the reason that the app accesses the user's location, this used while the app is running in the foreground.
```
Comment (breautek) [inline README.md]:
```suggestion
 * `NSLocationWhenInUseUsageDescription` describes the reason that the app accesses the user's location while the app is running in the foreground.
```

File: README.md [hunk 2]
```diff
@@ -160,15 +160,27 @@ error, the `geolocationError` callback is passed a
...
+ * `NSLocationAlwaysAndWhenInUseUsageDescription` describes the reason that the app is requesting access to the user’s location information at all times, use this key if your iOS app accesses location information while running in the background and foreground.
```
Comment (breautek) [inline README.md]:
```suggestion
 * `NSLocationAlwaysAndWhenInUseUsageDescription` describes the reason that the app is requesting access to the user’s location information at all times. Use this key if your iOS app accesses location information while running in the background and foreground. 
```

File: README.md [hunk 3]
```diff
@@ -160,15 +160,27 @@ error, the `geolocationError` callback is passed a
 
  This plugins requires the following usage description:
 
- * `NSLocationWhenInUseUsageDescription` describes the reason that the app accesses the user's location. 
+ * `NSLocationWhenInUseUsageDescription` describes the reason that the app accesses the user's location, this used while the app is running in the foreground.
+ * `NSLocationAlwaysAndWhenInUseUsageDescription` describes the reason that the app is requesting access to the user’s location information at all times, use this key if your iOS app accesses location information while running in the background and foreground. 
+ * `NSLocationAlwaysUsageDescription` describes the reason that the app is requesting access to the user's location at all times, use this key if your iOS app accesses location information in the background, and you deploy to a target earlier than iOS 11. In that case, add both this key and `NSLocationAlwaysAndWhenInUseUsageDescription` to your app’s `Info.plist` file with the same message.
```
Comment (breautek) [inline README.md]:
```suggestion
 * `NSLocationAlwaysUsageDescription` describes the reason that the app is requesting access to the user's location at all times. Use this key if your iOS app accesses location information in the background. and you deploy to a target earlier than iOS 11. In that case, add both this key and `NSLocationAlwaysAndWhenInUseUsageDescription` to your app’s `Info.plist` file with the same message.
```

> In that case, add both this key and `NSLocationAlwaysAndWhenInUseUsageDescription` to your app’s `Info.plist` file with the same message.

Can you clarify this? Are we suppose to use `NSLocationAlwaysAndWhenInUseUsageDescription` on iOS >= 12?

Comment (mosababubakr) [reply README.md]: Hello,
Yes as apple mentioned in this link: https://developer.apple.com/documentation/bundleresources/information_property_list/nslocationalwaysandwheninuseusagedescription

File: README.md (hunk 1 above)
Comment (breautek) [inline README.md]:
```suggestion
 * `NSLocationWhenInUseUsageDescription` describes the reason that the app accesses the user's location, this is used while the app is running in the foreground.
```

File: README.md (hunk 3 above)
Comment (breautek) [inline README.md]:
```suggestion
 * `NSLocationAlwaysUsageDescription` describes the reason that the app is requesting access to the user's location at all times. Use this key if your app accesses location information in the background and you deploy to a target earlier than iOS 11. For iOS 11 and later, add both `NSLocationAlwaysUsageDescription` and `NSLocationAlwaysAndWhenInUseUsageDescription` to your app’s `Info.plist` file with the same message.
```

File: README.md (hunk 1 above)
Comment (breautek) [reply README.md]: This suggestion is for grammar.

File: README.md (hunk 3 above)
Comment (breautek) [reply README.md]: This suggestion removes a run-on sentence and uses less "variables" such as "this" and "that" for clarity.

Comment (mosababubakr) [reply README.md]: You are wrong, “earlier” mean < iOS 11  mean also older (not newer).

And also , this used is better than this is used in that sentence.

Regards

Comment (breautek) [reply README.md]: > “this used “ is better than “this is used “ in that sentence.

That sounds weird to me when speaking it out, but grammar check tells me that you are right and I am wrong... so you can ignore this suggestion.

According to https://developer.apple.com/documentation/bundleresources/information_property_list/nslocationalwaysandwheninuseusagedescription

`NSLocationAlwaysAndWhenInUseUsageDescription` only applies to iOS 11+ and `NSLocationAlwaysUsageDescription` only applies to iOS 8-10

https://developer.apple.com/documentation/bundleresources/information_property_list/nslocationalwaysusagedescription

Am I misunderstanding something?

Comment (mosababubakr) [reply README.md]: Everything is good, thanks for info

**Comment by mosababubakr** [comment]
Please review it

File: README.md (hunk 2 above)
Comment (breautek) [reply README.md]: What do you think of this to break up a long-running sentence?

File: README.md (hunk 1 above)
Comment (mosababubakr) [reply README.md]: Seems good yes

**Comment by mosababubakr** [comment]
Please review it now and check if everything is ok

**Comment by patrizi** [comment]
Hi, 
I report that the problem is back. 

This is what Apple write me: 
> ITMS-90683: Missing Purpose String in Info.plist- Your app's code references one or more APIs that access sensitive user data. The app's Info.plist file should contain a NSLocationAlwaysUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. Starting Spring 2019, all apps submitted to the App Store that access user data are required to include a purpose string. If you're using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. You can contact the developer of the library or SDK and request they release a version of their code that doesn't contain the APIs. Learn more (https://developer.apple.com/documentation/uikit/core_app/protecting_the_user_s_privacy).

My configuration is: 
platform ios 5.1.1
cordova-plugin-geolocation 4.0.2 "Geolocation"

**Pull request merged**